pkg/generate/crds/gen.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crds
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	extensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/gengo/args"
+	"sigs.k8s.io/controller-tools/pkg/generate"
+	"sigs.k8s.io/controller-tools/pkg/internal/codegen"
+	"sigs.k8s.io/controller-tools/pkg/internal/codegen/parse"
+)
+
+type CRDGenerator struct {
```

Comment:
Given that it is under `crd` pkg, we can remove the prefix `CRD`.

Add comment explaining what it is for.